packages/api/internal/handlers/sandbox_create.go

```diff
@@ -233,12 +232,7 @@ func (a *APIStore) PostSandboxes(c *gin.Context) {
 
 	telemetry.ReportEvent(ctx, "Created analytics event")
 
-	go func() {
-		err = a.db.UpdateEnvLastUsed(context.Background(), env.TemplateID)
-		if err != nil {
-			a.logger.Errorf("Error when updating last used for env: %s", err)
-		}
-	}()
+	a.templateSpawnCounter.IncreaseTemplateSpawnCount(env.TemplateID)
```

Comment:
do this in go routine, you use lock for the flush, it would block spawning new sandboxes